src/main/java/frc/team2412/robot/util/DrivebaseWrapper.java

```diff
@@ -0,0 +1,99 @@
+package frc.team2412.robot.util;
+
+import edu.wpi.first.math.Matrix;
+import edu.wpi.first.math.estimator.SwerveDrivePoseEstimator;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.numbers.N1;
+import edu.wpi.first.math.numbers.N3;
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import swervelib.SwerveDrive;
+
+/**
+ * Wrapper class around members of {@link SwerveDrive}, like the Field2d telemetry and the pose
+ * estimator.
+ */
+public class DrivebaseWrapper {
+	@FunctionalInterface
+	private static interface VisionMeasurementAdder {
+		void addVisionMeasurement(Pose2d robotPose, double timestamp, Matrix<N3, N1> stdDevs);
+	}
+
+	private final Field2d field;
+
+	// Use the pose estimator ONLY for thread-safe operations!
+	private final SwerveDrivePoseEstimator poseEstimator;
+
+	// Since adding a vision measurement is not thread-safe, use a functional interface that can use
+	// SwerveDrive's methods (which are thread-safe)
+	private final VisionMeasurementAdder visionMeasurementAdder;
+
+	/** Constructor for when there is no drivebase. */
+	public DrivebaseWrapper() {
+		field = new Field2d();
+		// YAGSL will normally do this for us
+		// This may be pointless though since Field2d can't be shown in both SmartDashboard and the
+		// MatchDashboard tab
+		SmartDashboard.putData("Field", field);
```

Comment:
And per the comment here, this is logging the same field object here and in `MatchDashboard`, which seems potentially wasteful in bandwidth & processing.

Reply:
The intent was to make the robot code behave as similarly as possible with and without drivebase, and YAGSL will always put the field to SmartDashboard (which incidentally seems to be the only spot the sim GUI will render the field). It doesn't waste anything in bandwidth or processing, since the [`Field2d`](https://github.com/wpilibsuite/allwpilib/blob/main/wpilibj/src/main/java/edu/wpi/first/wpilibj/smartdashboard/Field2d.java) implementation only keeps track of the last spot it was put. (But it is weird to have it in two different places)

Reply:
Agree that it seems weird, but doesn't seem like a big problem so is fine to fix later.